### ISSUE TITLE
Update service catalog to 0.1.41

### ISSUE THREAD

**Issue by jasiu001** [kyma-project/kyma#1737]
AC:
- update Service Catalog to newer version than  0.1.40
- run test for service catalog, from https://github.com/kyma-incubator/service-catalog-tester.
- check the status https://github.com/kubernetes-incubator/service-catalog/issues/2521
- [This PR](https://github.com/kubernetes-incubator/service-catalog/pull/2493/files) was merged. Now we can remove [this custom logic](https://github.com/kyma-project/bundles/blob/master/bundles/azure-service-broker-0.0.1/chart/azure-service-broker/templates/pre-delete-job.yaml) from azure broker (**do not forget to test it**)
- if update done adjust a name of the ticket to the given version, and create a new one for the next sprint

**Comment by mszostok** [comment]
Please in next update, change also the client libraries in all our components, e.g. BUC is using ServiceCatalog client in version [0.1.28](https://github.com/kyma-project/kyma/blob/master/components/binding-usage-controller/Gopkg.toml#L52) (sic!)

**Comment by adamwalach** [comment]
Related issue:
- https://github.com/kyma-incubator/service-catalog-tester/issues/7

**Comment by adamwalach** [comment]
Test are green:

```
{"level":"info","log":{"ID":"2fbe9529-6951-48d9-8821-be9190f802fe","message":"Test \"E2E ServiceCatalog Happy Path\" end with success [start time: 2019-02-27 08:13:33.455738092 +0000 UTC m=+0.107261800, duration: 1m39.069384109s]","service":"test:runner","time":"2019-02-27T08:15:12.525Z"}}
{"level":"info","log":{"message":"Throttle test E2E ServiceCatalog Happy Path for 1m0s","service":"test:runner","time":"2019-02-27T08:15:12.525Z"}}
{"level":"info","log":{"ID":"532821a2-79ee-46b4-a6c9-35d4fd2cb130","message":"Starting test \"E2E ServiceCatalog Happy Path\"","service":"test:runner","time":"2019-02-27T08:16:12.525Z"}}
{"level":"info","log":{"ID":"532821a2-79ee-46b4-a6c9-35d4fd2cb130","message":"Test \"E2E ServiceCatalog Happy Path\" end with success [start time: 2019-02-27 08:16:12.525357041 +0000 UTC m=+159.176880750, duration: 1m18.558259953s]","service":"test:runner","time":"2019-02-27T08:17:31.083Z"}}
{"level":"info","log":{"message":"Throttle test E2E ServiceCatalog Happy Path for 1m0s","service":"test:runner","time":"2019-02-27T08:17:31.083Z"}}
{"level":"info","log":{"ID":"19c74338-b7f8-44df-ba34-7efc0073fc16","message":"Starting test \"E2E ServiceCatalog Happy Path\"","service":"test:runner","time":"2019-02-27T08:18:31.141Z"}}
{"level":"info","log":{"ID":"19c74338-b7f8-44df-ba34-7efc0073fc16","message":"Test \"E2E ServiceCatalog Happy Path\" end with success [start time: 2019-02-27 08:18:31.141737434 +0000 UTC m=+297.793261155, duration: 1m12.978145954s]","service":"test:runner","time":"2019-02-27T08:19:44.119Z"}}
{"level":"info","log":{"message":"Throttle test E2E ServiceCatalog Happy Path for 1m0s","service":"test:runner","time":"2019-02-27T08:19:44.119Z"}}
```

**Comment by adamwalach** [comment]
This issue is fixed: kubernetes-incubator/service-catalog#2521
